src/coreclr/vm/nativelibrary.cpp

```diff
@@ -654,13 +663,22 @@ namespace
         // even if it has one, or to leave off a prefix like "lib" even if it has one
         // (both of these are typically done to smooth over cross-platform differences).
         // We try to dlopen with such variations on the original.
-        const WCHAR* prefixSuffixCombinations[MaxVariationCount] = {};
+        NameVariations prefixSuffixCombinations[MaxVariationCount] = {};
         int numberOfVariations = ARRAY_SIZE(prefixSuffixCombinations);
         DetermineLibNameVariations(prefixSuffixCombinations, &numberOfVariations, wszLibName, libNameIsRelativePath);
         for (int i = 0; i < numberOfVariations; i++)
         {
             SString currLibNameVariation;
-            currLibNameVariation.Printf(prefixSuffixCombinations[i], PLATFORM_SHARED_LIB_PREFIX_W, wszLibName, PLATFORM_SHARED_LIB_SUFFIX_W);
+
+            NameVariations const variations = prefixSuffixCombinations[i];
+            if (variations | NameVariations_Prefix)
```

Comment:
```suggestion
            if ((variations & NameVariations_Prefix) != 0)
```
?

Reply:
Ugh. I'm sorry. This is not what I wanted to do. Blech. This is all wrong.